src/index.js

```diff
@@ -4,22 +4,21 @@ import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
-import { ThemeProvider } from "./contexts/theme/theme-context";
-import { LoginProvider, NotesProvider } from "contexts";
+import { FilterNotesProvider, LoginProvider, NotesProvider } from "contexts";
 
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
     <BrowserRouter>
-      <ThemeProvider>
-        <LoginProvider>
-          <NotesProvider>
+      <LoginProvider>
+        <NotesProvider>
+          <FilterNotesProvider>
             <App />
-          </NotesProvider>
-        </LoginProvider>
-      </ThemeProvider>
+          </FilterNotesProvider>
+        </NotesProvider>
+      </LoginProvider>
```

Comment:
And instead of adding the providers in index.js you can create a new file called as CombinedProviders.jsx

```js
import React from "react";
import { FilterNotesProvider, LoginProvider, NotesProvider } from "contexts";

const CombinedProvider = ({ children }) => {
    return (
         <LoginProvider>
        <NotesProvider>
          <FilterNotesProvider>
            {children}
          </FilterNotesProvider>
        </NotesProvider>
      </LoginProvider>
    )
}

export { CombinedProvider };
```
And then you can start using it in your index.js file just shifting your work from one file to another. 
So overall in index.js,this would be the view of index.js file

```js
ReactDOM.render(
 <React.StrictMode>
    <BrowserRouter>
       <CombinedProvider>
          <App>
      </CombinedProvider>
    </BrowserRouter>
  </React.StrictMode>,
  document.getElementById("root")
)
```

Reply:
I have left it as-is for now, maybe I will change this later.